pkg/controller/pxc/upgrade.go

```diff
@@ -302,6 +302,10 @@ func (r *ReconcilePerconaXtraDBCluster) smartUpdate(sfs api.StatefulApp, cr *api
 		return list.Items[i].Name > list.Items[j].Name
 	})
 
+	cr.Status.Status = api.AppStateInit
+	if err := r.writeStatus(cr); err != nil {
+		return errors.Wrap(err, "write status")
+	}
```

Comment:
nice move 👍 

I think we also need to update `cr.Status.PXC.Status`, `cr.Status.ProxySQL.Status` or `cr.Status.HAProxy.Status` depending on statefulset.